services/agents/mysql_test.go

```diff
@@ -107,3 +107,77 @@ func TestMySQLdExporterConfig(t *testing.T) {
 		assert.Equal(t, "DATA_SOURCE_NAME=tcp(1.2.3.4:3306)/?timeout=1s", actual.Env[0])
 	})
 }
+
+func TestMySQLdExporterConfigTablestatsGroupDisabled(t *testing.T) {
+	mysql := &models.Service{
+		Address: pointer.ToString("1.2.3.4"),
+		Port:    pointer.ToUint16(3306),
+	}
+	exporter := &models.Agent{
+		AgentID:                        "agent-id",
+		AgentType:                      models.MySQLdExporterType,
+		Username:                       pointer.ToString("username"),
+		Password:                       pointer.ToString("s3cur3 p@$$w0r4."),
+		TableCountTablestatsGroupLimit: -1,
+	}
+	actual := mysqldExporterConfig(mysql, exporter, redactSecrets)
+	expected := &agentpb.SetStateRequest_AgentProcess{
+		Type:               inventorypb.AgentType_MYSQLD_EXPORTER,
+		TemplateLeftDelim:  "{{",
+		TemplateRightDelim: "}}",
+		Args: []string{
+			"--collect.binlog_size",
+			"--collect.custom_query.hr",
+			"--collect.custom_query.hr.directory=/usr/local/percona/pmm2/collectors/custom-queries/mysql/high-resolution",
+			"--collect.custom_query.lr",
+			"--collect.custom_query.lr.directory=/usr/local/percona/pmm2/collectors/custom-queries/mysql/low-resolution",
+			"--collect.custom_query.mr",
+			"--collect.custom_query.mr.directory=/usr/local/percona/pmm2/collectors/custom-queries/mysql/medium-resolution",
+			"--collect.engine_innodb_status",
+			"--collect.engine_tokudb_status",
+			"--collect.global_status",
+			"--collect.global_variables",
+			"--collect.heartbeat",
+			"--collect.info_schema.clientstats",
+			"--collect.info_schema.innodb_cmp",
+			"--collect.info_schema.innodb_cmpmem",
+			"--collect.info_schema.innodb_metrics",
+			"--collect.info_schema.innodb_tablespaces",
+			"--collect.info_schema.processlist",
+			"--collect.info_schema.query_response_time",
+			"--collect.info_schema.userstats",
+			"--collect.perf_schema.eventsstatements",
+			"--collect.perf_schema.eventswaits",
+			"--collect.perf_schema.file_events",
+			"--collect.slave_status",
+			"--collect.standard.go",
+			"--collect.standard.process",
+			"--exporter.conn-max-lifetime=55s",
+			"--exporter.global-conn-pool",
+			"--exporter.max-idle-conns=3",
+			"--exporter.max-open-conns=3",
+			"--web.listen-address=:{{ .listen_port }}",
+		},
+		Env: []string{
+			"DATA_SOURCE_NAME=username:s3cur3 p@$$w0r4.@tcp(1.2.3.4:3306)/?timeout=1s",
+			"HTTP_AUTH=pmm:agent-id",
+		},
+		RedactWords: []string{"s3cur3 p@$$w0r4."},
+	}
+	requireNoDuplicateFlags(t, actual.Args)
+	require.Equal(t, expected.Args, actual.Args)
```

Comment:
if we check equality of `expected` and `actual` it's not required to check it for fields.